go/host/client_api_obscuro.go

```diff
@@ -53,27 +53,17 @@ func (api *ObscuroAPI) GetRollupHeader(hash gethcommon.Hash) *common.Header {
 
 // GetRollupHeaderByNumber returns the header for the rollup with the given number.
 func (api *ObscuroAPI) GetRollupHeaderByNumber(number *big.Int) (*common.Header, error) {
-	err := fmt.Errorf("no rollup with number %d is stored", number.Int64())
+	rollupHash := api.host.nodeDB.GetRollupHash(number)
 
-	// TODO - Provide a more efficient method on node DB to retrieve a rollup by number.
-	// We walk the chain back to the requested rollup.
-	rollupHeader := api.host.nodeDB.GetCurrentRollupHead()
-	for {
-		cmp := rollupHeader.Number.Cmp(number)
-		// We have found the rollup we're looking for.
-		if cmp == 0 {
-			return rollupHeader, nil
-		}
-		// The current rollup has a lower number than we are looking for. We stop walking the chain and return an error.
-		if cmp == -1 {
-			return nil, err
-		}
-
-		rollupHeader = api.host.nodeDB.GetRollupHeader(rollupHeader.ParentHash)
-		if rollupHeader == nil {
-			return nil, err
-		}
+	if rollupHash == nil {
+		return nil, fmt.Errorf("no rollup with number %d is stored", number.Int64())
 	}
+	rollupHeader := api.host.nodeDB.GetRollupHeader(*rollupHash)
```

Comment:
To minimise the increase in the data stored, we don't store a mapping from rollup number -> rollup header, but just from rollup number -> rollup hash, which can then be used to look up the header.